src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -365,6 +368,12 @@ private void ExtractToFileInternal(string filePath, string? linkTargetPath, bool
                 default:
                     throw new InvalidOperationException(string.Format(SR.TarEntryTypeNotSupportedForExtracting, EntryType));
             }
+
+            if (currentGlobalExtendedAttributes != null)
+            {
+                // TODO: pass it to the ExtractAs* methods or find any filesystem-related entries in the dictionary that should be applied on the extracted file if the extended attributes were not set
```

Comment:
This is the only thing left to address. Depending on the name attribute in the global extended attributes, I need to apply it to the file, if applicable.

Reply:
See my detailed answer here: https://github.com/dotnet/runtime/pull/69933#issuecomment-1146384242